types/persisted-model.d.ts

```diff
@@ -519,6 +520,67 @@ export declare class PersistedModel extends ModelBase {
    * @returns {String} The `id` property name
    */
   static getIdName(): string;
+
+  /**
+  * Register an asynchronous observer for the given operation (event).
+  *
+  * Example:
+  *
+  * Registers a `before save` observer for a given model.
+  *
+  * ```js
+  * MyModel.observe('before save', function filterProperties(ctx, next) {
+  if (ctx.options && ctx.options.skipPropertyFilter) return next();
+  if (ctx.instance) {
+    FILTERED_PROPERTIES.forEach(function(p) {
+      ctx.instance.unsetAttribute(p);
+    });
+  } else {
+    FILTERED_PROPERTIES.forEach(function(p) {
+      delete ctx.data[p];
+    });
+  }
+  next();
+  });
+  * ```
+  *
+  * @param {String} operation The operation name.
+  * @callback {function} listener The listener function. It will be invoked with
+  * `this` set to the model constructor, e.g. `User`.
+  */
+  static observe(operation: string, handler: Listener<OperationHookContext<PersistedModelClass>>): void;
+
+  /**
+  * Unregister an asynchronous observer for the given operation (event).
+  *
+  * Example:
+  *
+  * ```js
+  * MyModel.removeObserver('before save', function removedObserver(ctx, next) {
+    // some logic user want to apply to the removed observer...
+    next();
+  });
+  * ```
+  *
+  * @param {String} operation The operation name.
+  * @callback {function} listener The listener function.
+  */
+  static removeObserver(operation: string, handler: Listener<OperationHookContext<PersistedModelClass>>): void;
```

Comment:
I am a bit confused about this function, the function name gives me an impression that it removes a registered observer(registered by model.observe) by name, but it takes in a handler as the 2nd parameter...

Or does it remove ALL the observers bound to the operation?

Reply:
It removes all listeners for an operation.

This PR is not creating the method, though. It is just providing the TypeScript definitions for something that already existed. So any feedback on the looks and behavior of the method is for some other time and place.

Reply:
@jannyHou 

To register an Operation Hook observer, we need two pieces of information:
- what hook to observe (the operation name - a string like `before save`) and 
- the handler function to call. 

Similarly, to unregister an observer, we need the same two pieces.

Consider the following code:

```js
const logHook(ctx, next) {
  debug(ctx);
  next();
}

MyModel.observe('before save', logHook);
MyModel.observe('after save', logHook);

// now we want to remove 'after save' hook but preserve 'before save' hook
MyModel.removeObserver('after save', logHook);
```

Reply:
thanks @hacksparrow I see, didn't notice that the function already exists. 
and thanks for the explanation from @bajtos , very clear to me 👍